util/pyclient/bft_config.py

```diff
@@ -42,8 +42,8 @@ def __init__(self, n: int, f: int = 1, c: int = 0, clients=10):
         self.f : int = f
         self.c : int = c
         self.clients : int = clients
-        assert self.n == 3 * self.f + 2 * self.c + 1, \
-            f"Invariant breached. Expected: n = 3f + 2c + 1"
+        assert self.n >= 3 * self.f + 2 * self.c + 1, \
```

Comment:
Each replica has to have a role of either f or c.
Otherwise, this replica only increases the burden of the bft protocol without contributing to its resilience. Unless you have a good reason for this change I would remove it

Reply:
Since this code is used for testing purposes, I don't really care about the resiliency of the network, but rather if 5 replicas have a bug which 6 don't.